torch/csrc/jit/codegen/cuda/index_compute.cpp

```diff
@@ -2166,7 +2166,8 @@ std::vector<Val*> Index::getProducerStridedIndices(
   }
 
   std::vector<Val*> strided_indices;
-  if (producer->getMemoryType() == MemoryType::Global) {
+  if (producer->getMemoryType() == MemoryType::Global ||
+      ir_utils::isRandOp(consumer->definition())) {
     strided_indices =
         getGlobalProducerStridedIndices(producer, consumer, loops);
```

Comment:
I think we would need a `getLinearIndices`, which is similar to `getGlobalProducerStridedIndices` but always assume contiguous?